app/code/core/Mage/Index/Model/Lock.php

```diff
@@ -287,6 +288,10 @@ protected function _getLockFile($lockName)
             } else {
                 self::$_lockFileResource[$lockName] = fopen($file, 'x');
             }
+            if (!self::$_lockFileResource[$lockName]) {
+                self::$_lockFileResource[$lockName] = null;
```

Comment:
is this line necessary in case somebody catches the exception and continues the flow?

Reply:
I think the `null` is necessary because of the condition at line 283, it will think the lock file resource is valid.

Reply:
sure but there's a thrown exception on 293 so all the flow is exited

Reply:
This appears to be a singleton class, so if `fopen` fails, the next call to the same method with the same `$lockName` will return an invalid resource (`false`), unless the condition on line 283 is changed.